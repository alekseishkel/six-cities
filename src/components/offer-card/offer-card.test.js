import React from 'react';
import renderer from 'react-test-renderer';

import offersCardsInfo from '../mocks/offers';
import OfferCard from './offer-card.jsx';

it(`OfferPage correctly renders after relaunch`, () => {
  const tree = renderer.create(
      <OfferCard
        func={jest.fn()}
        cardInfo={offersCardsInfo[0]}
        currentActiveCard={jest.fn()}
      />
  );

  expect(tree).toMatchSnapshot();
});
