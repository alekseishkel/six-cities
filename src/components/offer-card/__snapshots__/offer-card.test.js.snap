// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferCard correctly renders after relaunch 1`] = `
<article
  className="favorites__card place-card"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <a
    href="/Hamburg/1"
    onClick={[Function]}
  >
    <div
      className="favorites__image-wrapper place-card__image-wrapper"
    >
      <img
        alt="Place image"
        className="place-card__image"
        height={150}
        src="https://htmlacademy-react-2.appspot.com/six-cities/static/hotel/7.jpg"
        width={150}
      />
    </div>
  </a>
  <div
    className="favorites__card-info place-card__info"
  >
    <div
      className="place-card__price-wrapper"
    >
      <div
        className="place-card__price"
      >
        <b
          className="place-card__price-value"
        >
          €
          482
        </b>
        <span
          className="place-card__price-text"
        >
          / night
        </span>
      </div>
      <button
        className="place-card__bookmark-button place-card__bookmark-buttonnull button"
        onClick={[Function]}
        type="button"
      >
        <svg
          className="place-card__bookmark-icon"
          height={19}
          width={18}
        >
          <use
            xlinkHref="#icon-bookmark"
          />
        </svg>
        <span
          className="visually-hidden"
        >
          In bookmarks
        </span>
      </button>
    </div>
    <div
      className="place-card__rating rating"
    >
      <div
        className="place-card__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "55.99999999999999%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <h2
      className="place-card__name"
    >
      The Pondhouse - A Magical Place
    </h2>
    <p
      className="place-card__type"
    >
      hotel
    </p>
  </div>
</article>
`;
